hibernate-envers/src/main/java/org/hibernate/envers/boot/internal/EnversServiceImpl.java

```diff
@@ -151,17 +153,22 @@ private void doInitialize(
 				revInfoCfgResult.getRevisionInfoTimestampData(),
 				serviceRegistry
 		);
-		this.entitiesConfigurations = new EntitiesConfigurator().configure(
-				metadata,
-				serviceRegistry,
-				reflectionManager,
-				mappingCollector,
-				globalConfiguration,
-				auditEntitiesConfiguration,
-				auditStrategy,
-				revInfoCfgResult.getRevisionInfoXmlMapping(),
-				revInfoCfgResult.getRevisionInfoRelationMapping()
-		);
+		this.entitiesConfigurations = AccessController.doPrivileged( new PrivilegedAction<EntitiesConfigurations>() {
```

Comment:
Couldn't this call be more fine grained? It sounds weird we need to run all the configure code in a privileged block?